src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -933,6 +938,14 @@ internal void ProcessHandshakeSuccess()
             Debug.Assert(_maxDataSize > 0, "_maxDataSize > 0");
 
             SslStreamPal.QueryContextConnectionInfo(_securityContext!, ref _connectionInfo);
+#if DEBUG
+            if (NetEventSource.Log.IsEnabled())
+            {
+                // This keeps the propery alive for tests via reflection
```

Comment:
What happens when tests run in release?

Reply:
yes. I was wondering if I should make common for both flavors as it may help debug handshake issues but at the end I decided to not to so we do not increase SslStream size. (even if the bool/int is small)